### ISSUE TITLE
폐렴 고찰

### ISSUE THREAD

**Issue by UGeunJi** [UGeunJi/Deep_Learning_Project#4]
할만 한 파라미터 조정 (이전 issue에서 한 것과 겹치지 않는 것들)

- learning_rate 초기 설정과 조절 (clear)
- 데이터 로그 변환 (clear)
- loss 함수 설정 -> Binary Cross Entropy, Categorical Cross Entropy, Sparse Categorical Cross Entropy [코드 참고](https://cheris8.github.io/artificial%20intelligence/DL-Keras-Loss-Function/) (clear)
- 모델 가져오기(전이 학습)

예시)
```
def build_resnet50():
    resnet50 = ResNet50(weights='imagenet', include_top=False)

    input = Input(shape=(SIZE, SIZE, N_ch))
    x = Conv2D(3, (3, 3), padding='same')(input)
    
    x = resnet50(x)
    
    x = GlobalAveragePooling2D()(x)
    x = BatchNormalization()(x)
    x = Dropout(0.5)(x)
    x = Dense(256, activation='relu')(x)
    x = BatchNormalization()(x)
    x = Dropout(0.5)(x)

    # multi output
    output = Dense(2,activation = 'softmax', name='root')(x)
 

    # model
    model = Model(input,output)
    
    optimizer = Adam(lr=0.003, beta_1=0.9, beta_2=0.999, epsilon=0.1, decay=0.0)
    model.compile(loss='categorical_crossentropy', optimizer=optimizer, metrics=['accuracy'])
    model.summary()
    
    return model
```

---

상위권 코드들의 결과를 보면 recall이 높고, precision과 accuracy는 낮은 것을 확인할 수 있습니다.
원인은 오버피팅의 영향으로 보고 있고, 해결책은 finetuning이라고 생각합니다.

이미 오버피팅이 발생하지 않은 코드도 있지만, 오버피팅이 발생한 코드도 개선하려는 과정도 담을 수 있으면 좋을 것 같습니다.

---

[참고할 코드 1 ](https://www.kaggle.com/code/amyjang/tensorflow-pneumonia-classification-on-x-rays)
[참고할 코드 2](https://www.kaggle.com/code/sanwal092/intro-to-cnn-using-keras-to-predict-pneumonia)
[참고할 코드 3](https://www.kaggle.com/code/rajmehra03/a-comprehensive-guide-to-transfer-learning)
[참고할 코드 4](https://www.kaggle.com/code/madz2000/pneumonia-detection-using-cnn-92-6-accuracy)
